### PR TITLE
Prevent sharing tokens in multi-tenanted applications.

### PR DESCRIPTION
**Pull request by quicken**
The default behaviour of the context object for using tokens for creating requests to the USI API is to share generated tokens. 

However, this causes unexpected authentication errors when the sample code is used in applications where multiple tenants need to make requests to the USI API.

For example suppose we have two tenants, Tenant A and Tenant B making requests to the USI.  If "tenant A" makes a request a token is generated and cached and the request succeeds. When tenant B subsequently attempts to call the USI the cached token from tenant A will be used from the cache causing the request to fail. 

In other words, the first tenant to call the USI will succeed and requests from other tenants will fail. This even happens if a new instance of a context object is created. 

To prevent this issue from occurring in a multi-tenanted use case the request context must be configured to NOT share tokens between requests. 

I have made the assumption that this would be the case in cloud applications as is reflected in this pull request.

### CODE REVIEW

**Review by bartland — APPROVED** [review]

**Comment by bartland** [comment]
@quicken  Thank you.
